src/snippet.rs

```diff
@@ -207,8 +207,8 @@ impl<'a, T: Clone> Snippet<'a, T> {
     /// not allowed to be passed to this function.
     ///
     /// </div>
-    pub fn origin(mut self, origin: &'a str) -> Self {
-        self.origin = Some(origin);
+    pub fn origin(mut self, origin: impl Into<Origin<'a>>) -> Self {
+        self.origin = Some(origin.into());
         self
```

Comment:
Not too sure how this makes it clearer.

Also, wht does uit mean to specify line, char_column, or primary on an Origin in a Snippet?

Reply:
`is_primary` is respected unless the `Origin` is the "primary origin" for a `Group` (I think).

`line` is respected if it is set.

`char_column` is respected if `line` is set; if `line` is not set, it is overridden.